nvtabular/dask/io.py

```diff
@@ -239,27 +262,48 @@ class ParquetDatasetEngine(DatasetEngine):
         Dask-based version of cudf.read_parquet.
     """
 
-    def __init__(self, *args, row_groups_per_part=None):
+    def __init__(self, *args, row_groups_per_part=None, legacy=False):
         # TODO: Improve dask_cudf.read_parquet performance so that
         # this class can be slimmed down.
         super().__init__(*args)
-        self._metadata, self._base = self.get_metadata()
-        self._pieces = None
+
+        if pa_ds and not legacy:
+            # Use pyarrow.dataset API for "newer" pyarrow versions.
+            # Note that datasets API cannot handle a directory path
+            # within a list.
+            if len(self.paths) == 1 and self.fs.isdir(self.paths[0]):
+                self.paths = self.paths[0]
+            self._legacy = False
+            self._pieces = None
+            self._metadata, self._base = defaultdict(int), ""
+            path0 = None
+            ds = pa_ds.dataset(self.paths, format="parquet")
+            # TODO: Allow filtering while accessing fragments.
+            #       This will require us to specify specific row-group indices
+            for file_frag in ds.get_fragments():
+                if path0 is None:
+                    path0 = file_frag.path
+                for rg_frag in file_frag.get_row_group_fragments():
```

Comment:
It seems pyarrow already replaced `get_row_group_fragments` with `split_by_row_group`... I guess using pyarrow.dataset is definitely living "on the bleeding edge" at the moment.

Reply:
Should we update here before merging?